.github/workflows/gradle-package.yml

```diff
@@ -0,0 +1,71 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will build a package using Gradle and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#Publishing-using-gradle
+
+name: Gradle Package
+
+on:
+  push:
+    branches: [ feature/**,develop ]
+  pull_request:
+    types: [ opened, reopened, synchronize ]
+  pull_request_review:
+    types: [ submitted, edited ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/BuildAction
+      - uses: ./.github/actions/JacocoBadge
+      - uses: ./.github/actions/TestAndCoverageReportAction
+
+      - name: Get artifact name
+        id: get_artifact_name
+        run: |
+          artifactName=$(./gradlew printVersion | grep '[0-9].[0-9].[0-9]-SNAPSHOT')
+          echo $artifactName
+          echo "##[set-output name=artifactName;]${artifactName}"
+
+      - name: Step 3 - Use the Upload Artifact GitHub Action
+        uses: actions/upload-artifact@v2
+        with:
+          name: "currency-rate-converter-${{ steps.get_artifact_name.outputs.artifactName }}.jar"
+          path: "build/libs/currency-rate-converter-${{ steps.get_artifact_name.outputs.artifactName }}.jar"
+
+  create_and_publish_docker_image:
+    runs-on: ubuntu-latest
+    needs: [ build ]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: currency-rate-converter-2.0.0-SNAPSHOT.jar
```

Comment:
This uses the fixed version of the artifact, which means that this step will break as soon as the artifact version is changed. 

How can you make this dynamic?